test/src/test/java/org/corfudb/integration/MultiStreamMergeStreamListenerIT.java

```diff
@@ -0,0 +1,300 @@
+        store.subscribeLrMergeListener(listenerBufferDefined, namespace, defaultTag,
+                Arrays.asList(defaultTableName), CorfuStoreMetadata.Timestamp.getDefaultInstance(),
+                deltaStreamBufferSize);
+
+        final int numUpdates = 10;
+        // Write updates to table_A only
+        writeUpdatesToDefaultTable(numUpdates, 0);
+
+        // Give time for deltas to be processed
+        TimeUnit.MILLISECONDS.sleep(sleepTime);
+        LinkedList<CorfuStreamEntries> updates = listener.getUpdates();
+        assertThat(updates.size()).isEqualTo(numUpdates);
+
+        // Write more updates to table_A and LrStatus table
+        writeUpdatesToBothTable(numUpdates/2, numUpdates);
+        TimeUnit.MILLISECONDS.sleep(sleepTime);
+        assertThat(updates.size()).isEqualTo(numUpdates + numUpdates/2);
+        verifyUpdates(updates);
+
+        assertThat(shutdownCorfuServer(corfuServer)).isTrue();
+
+        assertThat(listenerBufferDefined.updates.size()).isEqualTo(numUpdates + numUpdates/2);
+    }
+
+
+    /*
+     * This tests the scenario where streamListener is subscribed after initial write to one table, the
+     * other table has no data at this point. After sometime, data gets written to the same table.
+     * Verifies that the streamListener sees the updates from one table in the correct sequence.
+     * */
+    @Test
+    public void testStreamingClientTableFromTsCorfuTableEmpty() throws Exception {
+        // Run a corfu server & initialize CorfuStore
+        initializeCorfu();
+        openTable();
+
+        final int numUpdates = 10;
+        final int indexDefault= 0;
+        // Write updates to table_A only
+        writeUpdatesToDefaultTable(numUpdates, 0);
+
+        CorfuStoreMetadata.Timestamp snapshotTs;
+        SampleSchema.Uuid firstKey = SampleSchema.Uuid.newBuilder().setMsb(indexDefault).setLsb(indexDefault).build();
+        SampleSchema.SampleTableAMsg firstValue = SampleSchema.SampleTableAMsg.newBuilder().setPayload(String.valueOf(indexDefault)).build();
+
+        // read tail of table_A
+        try (TxnContext txn = store.txn(namespace)) {
+            CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.SampleTableAMsg, SampleSchema.Uuid> entry = txn.getRecord(defaultTableName, firstKey);
+            assertThat(entry.getPayload()).isEqualTo(firstValue);
+            snapshotTs = txn.commit();
+            assertThat(snapshotTs.getSequence()).isEqualTo(runtime.getSequencerView()
+                    .query(CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(namespace, defaultTableName))));
+        }
+
+        // Subscribe from latest Ts. The listener starts listening to both the LrStatus table and table_A from the Ts onwards.
+        StreamListenerImplMultiStream listener = new StreamListenerImplMultiStream();
+        store.subscribeLrMergeListener(listener, namespace, defaultTag, Arrays.asList(defaultTableName), snapshotTs);
```

Comment:
if snapshotTs == the last address in Table A's address space, should we get the previous(last) update on that table?